### PR TITLE
[wasm][debugger] Rebind invalid breakpoints after receiving an update in that source file

### PR DESCRIPTION
**Pull request by thaystg**
- moving hotreload debugger tests to another file
- sending Debugger.scriptParsed to browser when we receive an update in that file
- trying to rebind invalid breakpoints after receiving an update in that source file
- skip rebind breakpoints that don't have offset changed.

### CODE REVIEW

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs [hunk 1]
```diff
@@ -835,6 +845,11 @@ private async Task<bool> SendBreakpointsOfMethodUpdated(SessionId sessionId, Exe
                 {
                     await SetBreakpoint(sessionId, context.store, req, true, token);
                 }
+                //trying to reset breakpoints that were not bound but are in the same file of a method update
+                if (req.Method == null && method.Info.SourceName.Equals(req.File))
+                {
+                    await SetBreakpoint(sessionId, context.store, req, true, token);
+                }
```
Comment (lewing) [inline src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs:852]: I think the whole loop can become something like
```suggestion
             foreach (var req in context.BreakpointRequests.Values)
            {
                if (req.TryResolve(method.Source))
                {
                    await SetBreakpoint(sessionId, context.store, req, true, token);
                }
            }
```

Comment (lewing) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs:852]: We could refactor the setbreakpoint logic to be more efficient after that by only updating on method if needed

**Comment by radical** [comment]
Make sure to add the new class name to https://github.com/dotnet/runtime/blob/3bce91ce82aa1efe42412ff6462155275dd05c1e/eng/testing/scenarios/WasmDebuggerTestsJobsList.txt so it gets run on helix.

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs (hunk 1 above)
Comment (thaystg) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs]: Done.

**Review by lewing — APPROVED** [review]

**Pull request merged**